lib/ansible/modules/system/authorized_key.py

```diff
@@ -499,6 +508,9 @@ def enforce_state(module, params):
             # rank here is the rank in the provided new keys, which may be unrelated to rank in existing_keys
             parsed_new_key = (parsed_new_key[0], parsed_new_key[1], parsed_options, parsed_new_key[3], parsed_new_key[4])
 
+        if comment is not None:
+            parsed_new_key = (parsed_new_key[0], parsed_new_key[1], parsed_new_key[2], comment)
```

Comment:
Need to add the original index to the end of this tuple, otherwise an out of bounds error will be thrown on line 608.